File: backend/src/main/java/de/snickit/rezeptebackend/controller/UserController.java
package de.snickit.rezeptebackend.controller;

import de.snickit.rezeptebackend.entities.User;
import de.snickit.rezeptebackend.entities.repositories.UserRepo;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
public class UserController {

    final
    UserRepo userRepo;

    public UserController(UserRepo userRepo) {
        this.userRepo = userRepo;
    }

    @CrossOrigin
    @RequestMapping(
            method = RequestMethod.GET,
            path = "/user",
            produces = MediaType.APPLICATION_JSON_VALUE
    )
    public List<User> getAllUsers(){
        return userRepo.findAll();
    }

    @CrossOrigin
    @RequestMapping(
            method = RequestMethod.POST,
            path = "/user",
            produces = MediaType.APPLICATION_JSON_VALUE,
            consumes = MediaType.APPLICATION_JSON_VALUE
    )
    public User addPerson(@RequestBody User person) {
        return userRepo.save(person);
    }
}

File: backend/src/main/kotlin/de/snickit/recipe/backend/controller/UserController.kt
package de.snickit.recipe.backend.controller

import de.snickit.recipe.backend.persistence.entities.User
import de.snickit.recipe.backend.persistence.repositories.UserRepo
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*

@RestController(value = "/user")
class UserController(private val userRepo: UserRepo) {

    @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
    fun allUsers(): MutableList<User> = userRepo.findAll()

    @CrossOrigin
    @PostMapping(produces = [MediaType.APPLICATION_JSON_VALUE], consumes = [MediaType.APPLICATION_JSON_VALUE])
    fun addPerson(@RequestBody person: User): User {
        return userRepo.save(person)
    }

}
